tests/unit/utils/test_metadata_utils.py

```diff
@@ -63,3 +65,47 @@ def test_sort_prog_descriptions_returns_default_value(self) -> None:
             }
             not_in_config_value = {'source': 'test', 'text': 'I am a test'}
             self.assertEqual(_sort_prog_descriptions(mock_config, not_in_config_value), len(mock_config))
+
+
+class UneditableTableDescriptionTest(unittest.TestCase):
+    def setUp(self) -> None:
+        pass
+
+    def test_table_desc_match_rule_schema_only(self) -> None:
+        with local_app.app_context():
+            # Mock match rule, table name and schema
+            test_match_rule = MatchRuleObject(schema_regex=r"^(schema1)")
+
+            # assert result for given schema and match rule
+            self.assertEqual(_parse_editable_rule(test_match_rule, 'schema1', 'test_table'), False)
+            self.assertEqual(_parse_editable_rule(test_match_rule, 'schema2', 'test_table'), True)
+
+    def test_table_desc_match_rule_table_only(self) -> None:
+        with local_app.app_context():
+            # Mock match rule, table name and schema
+            test_match_rule = MatchRuleObject(table_name_regex=r"^noedit_([a-zA-Z_0-9]+)")
+
+            # assert result for given table name and match rule
+            self.assertEqual(_parse_editable_rule(test_match_rule, 'schema', 'noedit_test_table'), False)
+            self.assertEqual(_parse_editable_rule(test_match_rule, 'schema', 'editable_test_table'), True)
+
+    def test_table_desc_match_rule_multiple_schemas(self) -> None:
```

Comment:
nit: I'd say this test is not necessary. Our application logic doesn't place any logic on the regex itself, we just apply it and expect it to work. If we unit test the same exact logic with a different regex, we are essentially testing that the regex works as expected -- which we don't need to do because that is job of the `re` library.

Reply:
yes I see your point. Thanks for pointing out!